### PR TITLE
Made the about us page dynamic. Loads details from a json file.

### LINKED ISSUE #47: Team page; static data
Move the details of 'About us' page to a json file, and add the data to layouts using that json file.
Add details of members using list view, right now it is static.
Through this, adding of new members will be easier.

### PR DESCRIPTION
**Pull request by Kartikdot**
Team page; static data 
Fixes #47
Please review the changes. Gradle version has not been updated.

### CODE REVIEW

File: app/src/main/java/nsit/app/com/nsitapp/AboutUs2.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
+package nsit.app.com.nsitapp;
```
Comment (Swati4star) [inline app/src/main/java/nsit/app/com/nsitapp/AboutUs2.java]: Why have you changed the file name to AboutUs2? Let it be AboutUs only.

File: app/src/main/java/nsit/app/com/nsitapp/AboutUs2.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, Bundle savedInstanceState) {
+        rootView = inflater.inflate(R.layout.aboutus_activity, container, false);
+        ListView aboutUsListView = (ListView) rootView.findViewById(R.id.teamList);
+        aboutUsAct = getActivity();
+        View footerView = ((LayoutInflater) aboutUsAct.getApplicationContext().getSystemService(Context.LAYOUT_INFLATER_SERVICE)).inflate(R.layout.aboutus_footer, null, false);
+        mAdapter = new MemberAdapter(getActivity(), new ArrayList<AboutUsMember>());
+        aboutUsListView.setAdapter(mAdapter);
+        aboutUsListView.addFooterView(footerView);
+        TextView rep = (TextView) rootView.findViewById(R.id.GoToRepo);
+        TextView con = (TextView) rootView.findViewById(R.id.cont);
```
Comment (Swati4star) [inline app/src/main/java/nsit/app/com/nsitapp/AboutUs2.java]: Make the variable name more informative. rep & con doesn't actually tell, what these are for.

File: app/src/main/java/nsit/app/com/nsitapp/MainActivity.java [hunk 3]
```diff
@@ -61,6 +61,7 @@ protected void onCreate(Bundle savedInstanceState) {
         requestWindowFeature(Window.FEATURE_INDETERMINATE_PROGRESS);
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
+        appContext=getApplicationContext();
```
Comment (Swati4star) [inline app/src/main/java/nsit/app/com/nsitapp/MainActivity.java:64]: What is the need of this? I don't think you are using it anywhere.

File: app/src/main/java/nsit/app/com/nsitapp/aboutUs_loader.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+import android.content.Context;
+import android.support.v4.content.AsyncTaskLoader;
+
+import java.util.List;
+
+/**
+ * Created by Kartik Kwatra on 27-08-2017.
+ */
+
+public class aboutUs_loader extends AsyncTaskLoader<List<AboutUsMember>> {
```
Comment (Swati4star) [inline app/src/main/java/nsit/app/com/nsitapp/aboutUs_loader.java]: Class names should be like this..
"AboutUsLoader"

File: app/src/main/res/layout/aboutus_activity.xml [hunk 5]
```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (Swati4star) [inline app/src/main/res/layout/aboutus_activity.xml]: aboutUs is a fragment. Isn't it?

**Comment by Swati4star** [comment]
The functionality is complete overall. But there are a few changes, that needs to be taken care of, before merging. Also, Can you explain the reason for adding so many Java files? BAsically, tell the functionality of each one.

File: app/src/main/java/nsit/app/com/nsitapp/AboutUs2.java (hunk 2 above)
Comment (Kartikdot) [reply app/src/main/java/nsit/app/com/nsitapp/AboutUs2.java]: These variable names were picked up from the previous code. I'll change them.

**Comment by Kartikdot** [comment]
Thank you for the review. I have noted the changes to be made. 
On it! I'll also explain the functionality of the java files after making the required changes.

File: app/src/main/java/nsit/app/com/nsitapp/MainActivity.java (hunk 3 above)
Comment (Kartikdot) [reply app/src/main/java/nsit/app/com/nsitapp/MainActivity.java:64]: This is being used to get the json file using the getResources() method in class utility.

**Comment by Kartikdot** [comment]
Made the changes as per the review.

4 new Java files have been added and 1 modified. Here i list the functionality of each.

1. AboutUs - This is the main fragment. It sets the layout, initializes the adapter for the listview. It also implements Loader callbacks and calls the initLoader() method.

2. AboutUsMember - This class defines the properties of a team member which will be displayed on the about us page. Objects of this class are used to populate the list view.

3. AboutUsLoader - This is a custom loader. It extends the AsyncTaskLoader. I made a custom loader instead of directly using AsyncTaskLoader since in case the user rotates the phone, a new instance of AsyncTask would be created and a new thread would be started. This would result in wastage of memory. Also, repeated changes in orientation might start too many threads resulting the app to crash.

4. AboutUsUtility - This class fetches the list of team members from the json file and generates an ArrayList of the type AboutUsMember. Functions of this class are called on a separate thread using the loader. 

5. MemberAdapter - This class populates the list view using the ArrayList of AboutUsMember.

**Comment by Kartikdot** [comment]
Fixes #47

**Comment by Kartikdot** [comment]
Please merge #53

**Pull request merged**